homeassistant/components/assist_satellite/entity.py

```diff
@@ -36,11 +46,112 @@ class AssistSatelliteEntity(entity.Entity):
     entity_description: AssistSatelliteEntityDescription
     _attr_should_poll = False
     _attr_state: AssistSatelliteState | None = None
+    _attr_supported_features = AssistSatelliteEntityFeature(0)
 
     _conversation_id: str | None = None
     _conversation_id_time: float | None = None
 
-    _run_has_tts: bool = False
+    _is_announcing: bool = False
+    _tts_finished_event: asyncio.Event | None = None
+    _wake_word_future: asyncio.Future[str | None] | None = None
+
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added to hass."""
+        self._set_state(AssistSatelliteState.LISTENING_WAKE_WORD)
+
+    @property
+    def is_announcing(self) -> bool:
+        """Returns true if currently announcing."""
+        return self._is_announcing
+
+    async def async_announce(
+        self,
+        announce_text: str,
+        announce_media_id: str | None = None,
+        pipeline_entity_id: str | None = None,
+    ) -> None:
+        """Play an announcement on the satellite.
+
+        If announce_media_id is not provided, announce_text is synthesized to
+        audio with the selected pipeline.
+
+        Calls _internal_async_announce with media id and expects it to block
+        until the announcement is completed.
+        """
+        if self._is_announcing:
+            raise SatelliteBusyError
+
+        self._is_announcing = True
+
+        try:
+            if announce_media_id:
+                # Resolve to full URL
+                announce_media_id = async_process_play_media_url(
+                    self.hass, announce_media_id
+                )
+            else:
+                # Synthesize audio and get URL
+                pipeline_id = self._resolve_pipeline(pipeline_entity_id)
+                pipeline = async_get_pipeline(self.hass, pipeline_id)
+
+                tts_options: dict[str, Any] = {}
+                if pipeline.tts_voice is not None:
+                    tts_options[tts.ATTR_VOICE] = pipeline.tts_voice
+
+                tts_media_id = tts_generate_media_source_id(
+                    self.hass,
+                    announce_text,
+                    engine=pipeline.tts_engine,
+                    language=pipeline.tts_language,
+                    options=tts_options,
+                )
+                tts_media = await media_source.async_resolve_media(
+                    self.hass,
+                    tts_media_id,
+                    None,
+                )
+
+                # Resolve to full URL
+                announce_media_id = async_process_play_media_url(
+                    self.hass, tts_media.url
+                )
```

Comment:
If `announce_media_id` is provided and it is a media source URL, it should also be resolved and processed. I believe that this if-statement should be rewritten to something like:

```python
if not announce_media_id:
    …
    announce_media_id = tts_generate_media_source_id(…)

if media_source.is_media_source_id(annonuce_media_id):
    annonuce_media_id = await media_source.async_resolve_media(announce_media_id)

announce_media_id = async_process_play_media_url(
    self.hass, tts_media.url
)
```